app/controllers/application_controller.rb

```diff
@@ -18,17 +18,22 @@ def decoded_token
 
   def current_user
     return unless decoded_token
-
     @current_user ||= User.find_by(id: decoded_token["user_id"])
   end
 
   def authorize_user
     render json: { error: "Unauthorized" }, status: :unauthorized unless current_user
   end
 
-    def authorize_manager_or_owner!
+  def authorize_manager_or_owner!
     unless current_user&.role.in?(%w[manager owner])
       render json: { error: "Unauthorized" }, status: :unauthorized
     end
   end
+
+  def authorize_owner!
+    unless current_user&.super_user?
+      render json: { error: 'Unauthorized' }, status: :unauthorized
+    end
+  end
```

Comment:
I don't like how the owner and super_user are nested between one another.  However, this was the best way to currently execute this code.  It might need to be revisited in the future.